### PR TITLE
Release v0.32.0

### PR DESCRIPTION
**Pull request by djm81**
# Description

Release **v0.32.0** from `dev` to `main`. This PR includes the following merged changes:

- **ci-01** (v0.31.1): PR Orchestrator log artifacts — test logs, repro logs, and enforcement reports uploaded as workflow artifacts for failed-run debugging.
- **arch-06** (Enhanced module manifest security): Publisher and integrity metadata in `module-package.yaml`, checksum/signature verification, registration-time trust checks, signing automation, and docs. **Fixes** #208.
- **arch-07** (Schema extension system): `extensions` on Feature/ProjectBundle, `get_extension`/`set_extension`, `schema_extensions` in manifest, `ExtensionRegistry`. **Resolves** #213.

**Contract References**: arch-06 — `crypto_validator` (checksum/signature verification), registration trust checks in module loader; arch-07 — `get_extension`/`set_extension` with `@icontract`/`@beartype`, `ExtensionRegistry` validation.

## Type of Change

Please check all that apply:

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [x] 🔒 Contract enforcement (adding/updating `@icontract` decorators)
- [ ] 🧪 Test enhancement (scenario tests, property-based tests)
- [ ] 🔧 Refactoring (code improvement without functionality change)

## Contract-First Testing Evidence

**Required for all changes affecting CLI commands or public APIs:**

### Contract Validation

- [x] **Runtime contracts added/updated** (`@icontract` decorators on public APIs)
- [x] **Type checking enforced** (`@beartype` decorators applied)
- [x] **CrossHair exploration** completed: `hatch run contract-test-exploration`
- [x] **Contract violations** reviewed and addressed

### Test Execution

- [x] **Contract validation**: `hatch run contract-test-contracts` ✅
- [x] **Contract exploration**: `hatch run contract-test-exploration` ✅
- [x] **Scenario tests**: `hatch run contract-test-scenarios` ✅
- [x] **Full test suite**: `hatch run contract-test-full` ✅

### Test Quality

- [x] **CLI commands tested** with typer test client
- [x] **Edge cases covered** with Hypothesis property tests
- [x] **Error handling tested** with invalid inputs
- [x] **Rich console output verified** manually or with snapshots

## How Has This Been Tested?

**Contract-First Approach**: arch-06 and arch-07 were implemented with TDD; contracts and scenario tests validate manifest trust checks and extension get/set behavior. ci-01 changes are workflow-only and validated by PR orchestrator runs.

### Manual Testing

- [x] Tested CLI commands manually
- [x] Verified rich console output
- [x] Tested with different input scenarios
- [x] Checked error messages for clarity

### Automated Testing

- [x] Contract validation passes
- [x] Property-based tests cover edge cases
- [x] Scenario tests cover user workflows
- [x] All existing tests still pass

### Test Environment

- Python version: 3.11+
- OS: Linux (CI); local as per developer environment

## Checklist

- [x] My code follows the style guidelines (PEP 8, ruff format, isort)
- [x] I have performed a self-review of my code
- [x] I have added/updated contracts (`@icontract`, `@beartype`)
- [x] I have added/updated docstrings (Google style)
- [x] I have made corresponding changes to documentation
- [x] My changes generate no new warnings (basedpyright, ruff, pylint)
- [x] All tests pass locally
- [x] I have added tests that prove my fix/feature works
- [x] Any dependent changes have been merged (ci-01, arch-06, arch-07 merged to dev)

## Quality Gates Status

- [x] **Type checking** ✅ (`hatch run type-check`)
- [x] **Linting** ✅ (`hatch run lint`)
- [x] **Contract validation** ✅ (`hatch run contract-test-contracts`)
- [x] **Contract exploration** ✅ (`hatch run contract-test-exploration`)
- [x] **Scenario tests** ✅ (`hatch run contract-test-scenarios`)

## Screenshots/Recordings (if applicable)

N/A — release consolidation; CLI behavior documented in `docs/reference/module-security.md` and `docs/guides/extending-projectbundle.md`.

### CODE REVIEW

File: tests/unit/specfact_cli/registry/test_crypto_validator.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+from specfact_cli.registry.crypto_validator import (
+    verify_checksum,
+    verify_signature,
+)
+
+
+def test_checksum_verification_succeeds_when_values_match():
+    """When artifact checksum matches expected, verification SHALL pass."""
+    data = b"module artifact content"
+    expected = "sha256:" + "a" * 64  # will be replaced by actual hash in impl
```
[the remark this answers is not in the document]
Comment (djm81) [reply tests/unit/specfact_cli/registry/test_crypto_validator.py]: Fixed in next push
